cpukit/score/src/schedulerstrongapa.c

```diff
@@ -3,339 +3,820 @@
+
+    while ( curr_node !=  lowest_reachable ) {
+      curr_CPU = curr_strong_node->cpu_to_preempt;
+      next_node = _Scheduler_strong_APA_Get_scheduled( self, curr_CPU );
+      /* curr_node preempts the next_node; */
+      _Scheduler_SMP_Preempt(
+	context,
+	curr_node,
+	next_node,
+	_Scheduler_strong_APA_Allocate_processor
+      );
+
+      curr_node = next_node;
+      curr_strong_node = _Scheduler_strong_APA_Node_downcast( curr_node );
+    }
+
+    _Scheduler_strong_APA_Move_from_scheduled_to_ready( context, lowest_reachable );
+
+    needs_help = false;
+  } else {
+    needs_help = true;
+  }
+
+  /* Add it to Ready chain since it is now either scheduled or just ready. */
+  _Scheduler_strong_APA_Insert_ready( context,node, insert_priority );
+
+  return needs_help;
 }
 
-void _Scheduler_strong_APA_Block(
-  const Scheduler_Control *scheduler,
-  Thread_Control          *the_thread,
-  Scheduler_Node          *node
+/*
+ * BFS Algorithm for task arrival
+ * Enqueue node either in the scheduled chain or in the ready chain.
+ * node is the newly arrived node and is currently not scheduled.
+ */
+static inline bool _Scheduler_strong_APA_Enqueue(
+  Scheduler_Context *context,
+  Scheduler_Node    *node,
+  Priority_Control   insert_priority
 )
 {
-  Scheduler_Context *context = _Scheduler_Get_context( scheduler );
+  Scheduler_strong_APA_Context *self;
+  Scheduler_strong_APA_CPU     *CPU;
+  uint32_t	                cpu_max;
+  uint32_t              	cpu_index;
+  Per_CPU_Control              *cpu_to_preempt;
+  Scheduler_Node               *lowest_reachable;
+  Scheduler_strong_APA_Node    *strong_node;
 
-  _Scheduler_SMP_Block(
+  /* Denotes front and rear of the queue */
+  uint32_t	front;
+  uint32_t	rear;
+
+  front = 0;
+  rear = -1;
+
+  self = _Scheduler_strong_APA_Get_self( context );
+  strong_node = _Scheduler_strong_APA_Node_downcast( node );
+  cpu_max = _SMP_Get_processor_maximum();
+  CPU = self->CPU;
+
+  for ( cpu_index = 0 ; cpu_index < cpu_max ; ++cpu_index ) {
+    CPU[ cpu_index ].visited = false;
+
+    /* Checks if the thread_CPU is in the affinity set of the node */
+    if ( _Processor_mask_Is_set( &strong_node->Affinity, cpu_index ) ) {
+      Per_CPU_Control *cpu = _Per_CPU_Get_by_index( cpu_index );
+
+      if ( _Per_CPU_Is_processor_online( cpu ) ) {
+        rear = rear + 1;
+        CPU[ rear ].cpu = cpu;
+        CPU[ cpu_index ].visited = true;
+        CPU[ cpu_index ].preempting_node = node;
+      }
+    }
+  }
+
+  /*
+   * This assert makes sure that there always exist an element in the
+   * Queue when we start the queue traversal.
+   */
+  _Assert( !_Processor_mask_Is_zero( &strong_node->Affinity ) );
+
+  lowest_reachable = _Scheduler_strong_APA_Get_lowest_reachable(
+                       self,
+                       front,
+                       rear,
+                       &cpu_to_preempt
+                     );
```

Comment:
I'd have to look, do we actually indent these until beyond the assignment? I can see why you broke it differently above. This should be done consistently though...

Reply:
I made it consistent. There's no rule specified(https://docs.rtems.org/branches/master/eng/coding-formatting.html#eighty-character-line-limit) about beyond the assignment indents, but this looks better I believe.